components/feature/prompts/src/main/java/mozilla/components/feature/prompts/ext/BrowserStore.kt

```diff
@@ -0,0 +1,31 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.feature.prompts.ext
+
+import mozilla.components.browser.state.action.ContentAction
+import mozilla.components.browser.state.selector.findTabOrCustomTabOrSelectedTab
+import mozilla.components.browser.state.selector.selectedTab
+import mozilla.components.browser.state.store.BrowserStore
+import mozilla.components.concept.engine.prompt.PromptRequest
+import mozilla.components.feature.prompts.dialog.PromptDialogFragment
+import java.lang.ref.WeakReference
+
+internal fun BrowserStore.consumePromptFrom(
```

Comment:
Could you add a test as well? 🙃

Reply:
As you wish

Reply:
![external-content duckduckgo com](https://user-images.githubusercontent.com/1370580/120368069-9d733200-c2df-11eb-9b57-c89dcf1d3c60.gif)

r- for not using a gif.